### ISSUE TITLE
engge2 does not work on Windows

### ISSUE THREAD

**Issue by scemino** [scemino/engge2#2]
Hello!

Thanks!  Now it's better, but I have a lot of error messages.  My game version is wrong? 

Like:

" 
2022-12-05T09:39:21 | ERROR |
AN ERROR HAS OCCURRED [unknown]

2022-12-05T09:39:21 | ERROR |
CALLSTACK

2022-12-05T09:39:21 | ERROR | *FUNCTION [animateBirdObject()] StartScreen.bnut line [18]

2022-12-05T09:39:21 | ERROR |
LOCALS

2022-12-05T09:39:21 | ERROR | [delay] 0.36725503206253

2022-12-05T09:39:21 | ERROR | [dist] 2

2022-12-05T09:39:21 | ERROR | [this] TABLE

2022-12-05T09:39:22 | ERROR |
AN ERROR HAS OCCURRED [unknown]

2022-12-05T09:39:22 | ERROR |
CALLSTACK

2022-12-05T09:39:22 | ERROR | *FUNCTION [animateBirdObject()] StartScreen.bnut line [18]

2022-12-05T09:39:22 | ERROR |
LOCALS

2022-12-05T09:39:22 | ERROR | [delay] 0.97194802761078

2022-12-05T09:39:22 | ERROR | [dist] 2

2022-12-05T09:39:22 | ERROR | [this] TABLE

2022-12-05T09:39:22 | ERROR |
AN ERROR HAS OCCURRED [unknown]

2022-12-05T09:39:22 | ERROR |
CALLSTACK

2022-12-05T09:39:22 | ERROR | *FUNCTION [flashRadioLight()] StartScreen.bnut line [52]

2022-12-05T09:39:22 | ERROR |
LOCALS

2022-12-05T09:39:22 | ERROR | [this] TABLE

2022-12-05T09:39:22 | ERROR |
AN ERROR HAS OCCURRED [unknown]

2022-12-05T09:39:22 | ERROR |
CALLSTACK

2022-12-05T09:39:22 | ERROR | *FUNCTION [animateBirdObject()] StartScreen.bnut line [18]

2022-12-05T09:39:22 | ERROR |
LOCALS

2022-12-05T09:39:22 | ERROR | [delay] 1.0392628908157

2022-12-05T09:39:22 | ERROR | [dist] 2

2022-12-05T09:39:22 | ERROR | [this] TABLE

2022-12-05T09:39:23 | ERROR |
AN ERROR HAS OCCURRED [unknown]

2022-12-05T09:39:23 | ERROR |
CALLSTACK

2022-12-05T09:39:23 | ERROR | *FUNCTION [animateBirdObject()] StartScreen.bnut line [18]

2022-12-05T09:39:23 | ERROR |
LOCALS

2022-12-05T09:39:23 | ERROR | [delay] 2.4500908851624

2022-12-05T09:39:23 | ERROR | [dist] 2

2022-12-05T09:39:23 | ERROR | [this] TABLE

2022-12-05T09:39:24 | ERROR |
AN ERROR HAS OCCURRED [unknown]

2022-12-05T09:39:24 | ERROR |
CALLSTACK

2022-12-05T09:39:34 | INFO | curThread.id: 300009
2022-12-05T09:39:34 | INFO | add breakwhilecond name=breakwhilerunning(300010) pid=300009
2022-12-05T09:39:36 | ERROR |
AN ERROR HAS OCCURRED [unknown]

2022-12-05T09:39:36 | ERROR |
CALLSTACK

2022-12-05T09:39:36 | ERROR | *FUNCTION [unknown()] EasyHardMode.bnut line [29]

2022-12-05T09:39:36 | ERROR |
LOCALS

2022-12-05T09:39:36 | ERROR | [this] TABLE

"

_Originally posted by @xcom169 in https://github.com/scemino/engge2/issues/1#issuecomment-1336961839_

**Comment by scemino** [comment]
This issue is reproducible, create a file `ng.nut`with this minimal code
```squirrel
include("Globals.nut")
include("DefineSounds.nut")
include("AchievementHelpers.nut")
include("StartScreen.nut")
defineRoom(StartScreen)
cameraInRoom(StartScreen)
```

the issue occurs when updating the threads, squirrel threads are resumed but something bad is happening.

**Comment by arnaullv** [comment]
Was this issued ever fixed? I was trying to run it in Windows but I didn't have success I hope this project is some day integrated inside Scummvm

**Comment by Ohrwerk** [comment]
Sorry to be so blunt here, but why do we need this engine remake?
The game runs on modern systems. Or are there problems?

**Comment by eblanca** [comment]
Even when software itself is good, open software is better.
Look at https://github.com/scummvm/scummvm/pull/5662 as a C++ version of this engine has been merged into ScummVM.
Now, look at the platforms TWP officially supports and compare against how many OSs ScummVM can run on... everybody wins.

**Comment by Ohrwerk** [comment]
> Even when software itself is good, open software is better. Look at [scummvm/scummvm#5662](https://github.com/scummvm/scummvm/pull/5662) as a C++ version of this engine has been merged into ScummVM. Now, look at the platforms TWP officially supports and compare against how many OSs ScummVM can run on... everybody wins.

You're right, open source is better. Thimbleweed Park runs on Windows 7, Mac OS and Linux. It's a bit of a problem with MAC OS, since OS "Classic" there has often been the problem that older software is no longer supported. As a Windows user up to Windows XP, this was practically unheard of. I didn't buy Thimbleweed Park because I wasn't sure at the time. Delores only supports Windows 10, so I didn't want to run the risk of disappointment.
When I look at my game library, I'm very happy about powerful tools like SCUMMVM, I also like the different DOSBox variants or 86Box for unsupported games.

Sometimes I just don't think far enough, but I was probably just disappointed that Delores is not supported.
Especially for someone like me, who has been using Jasc Paint Shop Pro 9 for over 20 years, this shouldn't happen. I just can't get used to GIMP.

**Comment by xcom169** [comment]
The support is already in the daily Scummvm build?

**Comment by arnaullv** [comment]
Not yet, but you can test it yourself if you revert the commit that pulls out the support and compile it
